### DISCUSSION TITLE
Is it possible to stack images with a specific suffix?

### DISCUSSION THREAD

**Discussion by odedia** [photoprism/photoprism#4166] [General]
My new favorite tool OSXphotos can export edited JPEGs as an additional output, along for a suffix. For example _IMG1450.jpeg in Apple Photos will export the orignal jpeg, plus an additional _IMG1450_edited.jpeg (that _edited suffix is customizable).
Right now, PhotoPrism treats the edited photo as a separate file and doesn't add it to the same stack as the original.
Is there a way to tell PhotoPrism to stack images with the same name but an additional suffix?

**Comment by lastzero** [comment]
Could you also use `.` instead of `_` so that the suffix would be `.edited.jpeg`, for example? I'm also wondering why all your file names seem to start with `_`?

**Comment by odedia** [comment]
I’ll give it a try. Is there a way to designate which should be at the top of the stack?
The _ is just how files are named in some cameras (Nikon).

**Comment by odedia** [comment]
Doesn't seem to work.

The right image is the stacked original .NEF, the .JPG and the .XMP file describing the .JPG.
The left image is the .edited.JPG and its .XMP file.

As you can see - both images are on the exact same date and time but it didn't identify them correctly as belonging to the same stack.

<img width="745" alt="image" src="https://github.com/photoprism/photoprism/assets/2992180/00e2a570-6c5c-4756-bf19-911b06b57121">

**Reply by lastzero** [reply]
That was just an example... The actual naming options for copies of files (aka sequential naming) can be found in our official documentation and here in our public source code:
https://github.com/photoprism/photoprism/blob/572d1a59175b222f6c6a6896320f075672d12c52/pkg/fs/filepath.go#L15-L29

**Comment by odedia** [comment]
I tried to rename all .edited.jpeg to be .jpg, and all the non-edited jpegs to end with .jpeg.
This one seems to work, but it is ofcourse more confusing to know which is the edited and which is not.
If there's any way to add this feature, I'll open a feature request. I'm also not sure if this order will be preserved throughout (.jpg on top, .jpeg secondary) or if it's random.

<img width="395" alt="image" src="https://github.com/photoprism/photoprism/assets/2992180/4e65293f-9114-427e-9270-0a214e08a146">

<img width="1420" alt="image" src="https://github.com/photoprism/photoprism/assets/2992180/eeeae3da-c6a8-4b9b-ba29-e1677f817434">

**Comment by odedia** [comment]
Oddly, it does work for the below. Is that because the .XMP files share the same id? 
EDIT: the previous example also shares the same id int the XMP files so I'm not sure why they are behaving different. I also updated the original .NEF to have the exact timestamp as the edited JPEG, but it wouldn't stack them. 

<img width="1258" alt="image" src="https://github.com/photoprism/photoprism/assets/2992180/fe500c14-4059-43c6-a418-e540d3258fdc">

**Reply by graciousgrey** [reply]
The files from your screenshot might be stacked if you have stacking by place and time enabled and they share the exact same date/time and GPS coordinates.

If you want to group images based on an ID, you can set set the fields "ImageUniqueID" or the "InstanceID" with Exiftool.
We do not yet read these fields from xmp sidecar files.

https://www.photoprism.app/kb/metadata

Since some camera manufacturers or other tools use these fields for serial numbers or similar, we only group files if these fields contain a valid UUID, such as 550e8400-e29b-11d4-a716-446655440000.

**Comment by odedia** [comment]
Happy to share a sample photos for reference if that would help.
